src/odb/include/odb/db.h

```diff
@@ -7456,9 +7480,58 @@ class dbMetalWidthViaMap : public dbObject
   // User Code End dbMetalWidthViaMap
 };
 
+class dbModBTerm : public dbObject
+{
+ public:
+  const char* getName() const;
+
+  dbModule* getParent() const;
+
+  void setNet(dbModNet* net);
+
+  dbModNet* getNet() const;
+
+  void setNextNetModbterm(dbModBTerm* next_net_modbterm);
+
+  dbModBTerm* getNextNetModbterm() const;
+
+  void setPrevNetModbterm(dbModBTerm* prev_net_modbterm);
+
+  dbModBTerm* getPrevNetModbterm() const;
+
+  void setNextEntry(dbModBTerm* next_entry);
+
+  dbModBTerm* getNextEntry() const;
+
+  // User Code Begin dbModBTerm
+
+  void setSigType(dbSigType& type);
+  dbSigType getSigType();
+  void setIoType(dbIoType type);
+  dbIoType getIoType();
+
+  static dbModBTerm* create(dbModule* parentModule, const char* name);
+
+  void connect(dbModNet*);
+  void disconnect();
+
+  char* getName();
+
+  void staSetPort(void* p);
+  void* staPort();
+
+ private:
+  void setFlags(uint flags);
+  uint getFlags() const;
```

Comment:
private methods should be on the implementation class not the public class

Reply:
i wonder what that means. The implementation class is automatically generated so the only place to put it is on the parent class.  Or is there another approach ?

Reply:
You can have a // User Code section in the generated class